File: Solution/WebSolution/Dotnet/AssessmentPortal/Monolithic/Monolithic_Controller/AssessmentAPI/Controllers/AuthController.cs
using System.Diagnostics;
using Microsoft.AspNetCore.Mvc;

using Transflower.TFLAssessment.Entities;

using Transflower.TFLAssessment.Services.Interfaces;

namespace Transflower.TFLAssessment.Controllers;

[ApiController]
[Route("api/[controller]")]
public class AuthController : ControllerBase
{
    private readonly IAuthService _service;
    private readonly ILogger<AuthController> _logger;

    public AuthController(IAuthService service, ILogger<AuthController> logger)
    {
        _service = service;
        _logger = logger;
    }

    [HttpPost("login")]
    public async Task<IActionResult> Login([FromBody] LoginRequest request)
    {
        var response = await _service.GetUserWithRolesByEmail(request.Email, request.Password);
        if (response == null)
        {
            _logger.LogWarning("Login failed for {Email}", request.Email);
            return Unauthorized("Invalid credentials");
        }

        _logger.LogInformation("User {Email} logged in", request.Email);
        return Ok(response);
    }


    [HttpPost("register")]
        public async Task<IActionResult> Register([FromBody] User user)
        {
            try
            {
                if (user == null)
                    return BadRequest(new { message = "Invalid request body." });

                if (string.IsNullOrWhiteSpace(user.AadharId) ||
                    string.IsNullOrWhiteSpace(user.Firstname) ||
                    string.IsNullOrWhiteSpace(user.Lastname) ||
                    string.IsNullOrWhiteSpace(user.Email) ||
                    string.IsNullOrWhiteSpace(user.ContactNumber) ||
                    string.IsNullOrWhiteSpace(user.Password))
                {
                    return BadRequest(new { message = "All fields are required." });
                }

                 bool isAdded = await _service.AddUser(user);  // ✅ capture return

        if (isAdded)
        {
            Console.WriteLine("✅ User registered successfully!");
            return Ok(new { message = "User registered successfully!" });
        }
        else
        {
            Console.WriteLine("❌ User registration failed.");
            return StatusCode(500, new { message = "Failed to register user." });
        }
            }
            catch (Exception ex)
            {
                Console.WriteLine($" ERROR in Register: {ex.Message}");
                Console.WriteLine(ex.StackTrace);

                // Return detailed message so frontend can see it
                return StatusCode(500, new { message = $"Server error: {ex.Message}" });
            }
        }
        [HttpPost("change-password")]
        public async Task<IActionResult> ChangePassword([FromBody] ChangePasswordRequest request)
        {
        var result = await  _service.ChangePassword(request);

         if (!result)
        return BadRequest("Old password is incorrect or update failed.");

          return Ok("Password updated successfully.");
}



}

File: Solution/CoMentor/Integrated/nodejs/backend/src/controllers/authcontroller.js
const ChangePasswordDto = require("../dtos/requests/changepassworddto");
const Credential = require("../dtos/requests/credential");
const UserRequest = require("../dtos/requests/userrequest");
const LoginStatsResponseDto = require("../dtos/responses/loggerlogin-statsresponsedto");
const LoginStatus = require("../dtos/responses/loginstatus");
const ResponseGenerator = require("../helpers/ResponseGenerator");

class AuthController {
  constructor(authService) {
    this.service = authService;
  }

 
  validate (req, res) {
    const responseGenerator = new ResponseGenerator();
    const credential = new Credential(
      req.body.username,
      req.body.password,
      req.body.role,
    );
    this.service.validate(credential, (err, result) => {
      if (err) {
        return responseGenerator.sendError(res, "Server error", 500, err);
      }

      const isValid = Array.isArray(result) && result.length > 0;

      const loginStatus = new LoginStatus(
        isValid,
        isValid ? "login successfully" : "login failed",
        isValid ? result[0].id : 0,
      );

      if (isValid) {
        return responseGenerator.sendSuccess(
          res,
          loginStatus,
          200,
          "Validation successful",
        );
      } else {
        return responseGenerator.sendError(res, "Invalid credentials", 401);
      }
    });
  };

  register(req, res) {
    const responseGenerator = new ResponseGenerator();
    const user = new UserRequest(
      req.body.firstName,
      req.body.lastName,
      req.body.email,
      req.body.password,
      req.body.contact,
    );
    this.service.register(user, (err, result) => {
      responseGenerator.generateResponse(
        res,
        err,
        result,
        "Failed to register user",
        "User registered successfully",
      );
    });
  }

  changePassword(req, res) {
    const responseGenerator = new ResponseGenerator();
    const changePassword = new ChangePasswordDto(
      req.body.id,
      req.body.oldPassword,
      req.body.newPassword,
    );
    if (
      !changePassword.id ||
      !changePassword.oldPassword ||
      !changePassword.newPassword
    ) {
      return res.status(400).json({ message: "All fields are required" });
    }
    this.service.changePassword(changePassword, (err, result) => {
      responseGenerator.generateResponse(
        res,
        err,
        result,
        "Failed to change password",
        "Password changed successfully",
      );
    });
  }
}

module.exports = AuthController;
